tags-config-service-impl/src/main/java/org/hypertrace/tag/config/service/TagConfigServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package org.hypertrace.tag.config.service;
+
+import io.grpc.Channel;
+import io.grpc.stub.StreamObserver;
+import java.util.List;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+import org.hypertrace.tag.config.service.v1.CreateTag;
+import org.hypertrace.tag.config.service.v1.CreateTagRequest;
+import org.hypertrace.tag.config.service.v1.CreateTagResponse;
+import org.hypertrace.tag.config.service.v1.DeleteTagRequest;
+import org.hypertrace.tag.config.service.v1.DeleteTagResponse;
+import org.hypertrace.tag.config.service.v1.GetTagRequest;
+import org.hypertrace.tag.config.service.v1.GetTagResponse;
+import org.hypertrace.tag.config.service.v1.GetTagsRequest;
+import org.hypertrace.tag.config.service.v1.GetTagsResponse;
+import org.hypertrace.tag.config.service.v1.Tag;
+import org.hypertrace.tag.config.service.v1.TagConfigServiceGrpc;
+import org.hypertrace.tag.config.service.v1.UpdateTagRequest;
+import org.hypertrace.tag.config.service.v1.UpdateTagResponse;
+
+@Slf4j
+public class TagConfigServiceImpl extends TagConfigServiceGrpc.TagConfigServiceImplBase {
+  private final ConfigServiceCoordinator configServiceCoordinator;
+
+  public TagConfigServiceImpl(Channel configChannel) {
+    configServiceCoordinator = new ConfigServiceCoordinatorImpl(configChannel);
+  }
+
+  @Override
+  public void createTag(
+      CreateTagRequest request, StreamObserver<CreateTagResponse> responseObserver) {
+    RequestContext requestContext = RequestContext.CURRENT.get();
+    CreateTag createTag = request.getTag();
+    Tag tag =
+        Tag.newBuilder().setId(UUID.randomUUID().toString()).setKey(createTag.getKey()).build();
+    Tag createdTag = configServiceCoordinator.upsertTag(requestContext, tag);
+    responseObserver.onNext(CreateTagResponse.newBuilder().setTag(createdTag).build());
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void getTag(GetTagRequest request, StreamObserver<GetTagResponse> responseObserver) {
+    RequestContext requestContext = RequestContext.CURRENT.get();
+    String tagId = request.getId();
+    Tag tag = configServiceCoordinator.getTag(requestContext, tagId);
+    responseObserver.onNext(GetTagResponse.newBuilder().setTag(tag).build());
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void getTags(GetTagsRequest request, StreamObserver<GetTagsResponse> responseObserver) {
+    RequestContext requestContext = RequestContext.CURRENT.get();
+    List<Tag> tagList = configServiceCoordinator.getAllTags(requestContext);
+    responseObserver.onNext(GetTagsResponse.newBuilder().addAllTags(tagList).build());
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void updateTag(
+      UpdateTagRequest request, StreamObserver<UpdateTagResponse> responseObserver) {
+    RequestContext requestContext = RequestContext.CURRENT.get();
+    Tag updateTag = request.getTag();
```

Comment:
We should get tag to check if it exists, before update

Reply:
Okay, so what should be the behaviour if there is no tag with that id and key?

Reply:
I would say throw exception. Does this need to be public? The interface does not seem to include this.

Reply:
We need to return 404 status here. Let me check on that front.

Reply:
What about the behaviour of getTag(id) and if id is not present?

Reply:
Ignore comment about 404 status. I think config service will throw exception with Status.NOT_FOUND, and that is good enough.

Reply:
Config-Service throws an exception with Status.UNKNOWN.